### PR TITLE
zstd: do not package both static & shared on windows

### PR DESCRIPTION
**Pull request by SpaceIm**
fix regression from https://github.com/conan-io/conan-center-index/pull/18826 where both shared & static were packaged on Windows if shared=True (see https://github.com/conan-io/conan-center-index/pull/18826#issuecomment-1769360351)

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — APPROVED** [review]
Great catch @SpaceIm! :)

**Comment by SpaceIm** [comment]
Actually lib name can be hard to predict, so let's keep `collect_libs`, and if static lib has to be removed, we remove `*.a` and `*_static.*` files: https://github.com/facebook/zstd/blob/v1.5.5/build/cmake/lib/CMakeLists.txt#L119-L124

**Review by uilianries — APPROVED** [review]
Thank you for spotting it and providing a fix!

File: recipes/zstd/all/conanfile.py [hunk 1]
```diff
@@ -86,24 +86,25 @@ def package(self):
         rmdir(self, os.path.join(self.package_folder, "share"))
 
         if self.options.shared and self.options.build_programs:
```
Comment (uilianries) [inline recipes/zstd/all/conanfile.py:88]:
```suggestion
        if self.options.shared:
```

Programs are statically linked already, we don't need to package them anyway.

Comment (SpaceIm) [reply recipes/zstd/all/conanfile.py:88]: This branch removes static libs when you have both shared & build_programs options enabled, it's correct as it is. See generate() method.

File: recipes/zstd/all/conanfile.py [hunk 2]
```diff
@@ -86,24 +86,25 @@ def package(self):
...
-            #If we built programs we always build static libs,
-            #but if we only want shared libs in the package then remove the static libs
+            # If we built programs we always build static libs,
+            # but if we only want shared libs in the package then remove the static libs
```
Comment (SpaceIm) [inline recipes/zstd/all/conanfile.py:90]:
```suggestion
            # If we build programs we have to build static libs (see logic in generate()),
            # but if shared is True, we only want shared lib in package folder.
```

**Review by uilianries — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**